<?xml version="1.0" standalone="no" ?>

<MergedConfig>

<default>

	<ObjectRecognition>
		<!-- Objects to load at startup -->
		<value name="sLoadObjects" value=""/>
		<!-- Path where the object descriptors are stored. -->
                <value name="sDataPath" value="/objectProperties/"/>
		<!-- relative size of border added to object images -->
		<value name="fObjectImageBorder" value="0.1"/>
		<!-- Resolution of captured camera images. 1=full, 2=half, 4=quarter -->
		<value name="iCaptureScaling" value="1"/>
		<!-- Minimal number of matched features for recognized object -->
		<value name="iMinMatchedKeyPoints" value="6" />
		<!-- Minimal percentage of matched features for recognized object compared to scene features -->
		<value name="fMinMatchPercentage" value="0.06" />
		<!-- stage 1 matcher (1=nearest neighbor, 2=flann) -->
		<value name="iStage1Matcher" value="2" />
		<!-- stage 2 matcher (1=simple hough, 2=hough) -->
		<value name="iStage2Matcher" value="2" />
		<!-- Number of images in queue in continuous mode -->
		<value name="iMaxImagesInPipeline" value="1" />

		<Histogram> <!-- TODO: is this section still needed? -->
			<value name="fHistogramClearRange" value="0.03" min="0" max="1" />
			<!-- For Y values < minY or > maxY, the UV values are omitted -->
			<!-- Usually, YUV cameras have a Y range of 16..235 -->
			<value name="iMinY" value="5" />
			<value name="iMaxY" value="245" />
			<!-- Maximal difference of scene snippet and object histogram -->
			<value name="fMaxMeanDifference" value="1.0" />
			<!-- Value to be substracted from the scene histogram before dividing through it. Is multiplied by the mean value. -->
			<value name="fSceneOffset" value="0.0" /><!-- 0.3 -->
			<!-- If the threshold is lower, more colors are detected as object colors (1.0 means the mean value is used as threshold) -->
			<value name="fThreshold" value="0.01" /> <!-- 0.25 -->
			<!-- Size of the histogram bins. For values above zero, neighboring u and v values are treated as identical. -->
			<value name="iBinSize" value="4"/>
			<!-- Size of dilation filter for binary histogram -->
			<value name="fHistogramDilationRadius" value="0.0" />
			<!-- Radius of open filter for optimizing the backprojection matrix -->
			<value name="fOpenRadius" value="15.0" />
		</Histogram>

		<NNRMatching>
			<!-- Max. nearest neighbor ratio for matching feature vectors -->
			<value name="fMaxNearestNeighbourRatio" value="0.85" />
		</NNRMatching>

		<SimpleHoughClustering>
			<value name="fMaxMatchDistance" value="15.0" />
		</SimpleHoughClustering>

		<Homography>
		 <!-- The method used to compute homography matrix; one of the following:
					0 - regular method using all the point pairs
					1 - RANSAC-based robust method (default)
					2 - Least-Median robust method -->
			<value name="iMethod" value="1" />
		</Homography>

		<Ransac>
			<value name="fMaxReprojectionError" value="5.0" />
		</Ransac>

		<HoughClustering>
			<!-- Number of KeyPoints needed to keep cluster -->
			<value name="iMinMatchNumber" value="5" />
			<value name="iAccumulatorSearchStrategy" value="1" />
			<!-- default values : 10,10,10,10 = 10000 bins -->
			<!-- Number of bins used for scale -->
			<value name="iScaleBins" value="10" />
			<!-- Number of bins used for orientation -->
			<value name="iOrientationBins" value="10" />
			<!-- Number of bins used for x position -->
			<value name="iXLocationBins" value="10" />
			<!-- Number of bins used for y position -->
			<value name="iYLocationBins" value="10" />
			<!-- If true plots and histograms are plotted-->
			<value name="bPlot" value="0"/>
			<!-- Minimum variance to predict a maximum in accumulator; if set to 0 does nothing -->
			<value name="fVariance" value="0" />
		</HoughClustering>

	</ObjectRecognition>


	<KeyPointExtraction>
		<!-- 0=PanomaticSurf, 1=Original Implementation -->
		<value name="iAlgorithm" value="0"/>

		<!-- Default parameters for the SURF library -->
		<Surf>
			<!-- Use rotation invariant features. Requires more computation time and
					 results in less distinctive features, so deactivate when not needed.
					 Default: 1 -->
			<value name="bRotationInvariance" value="1"/>
			<!-- 3 for SURF-36, 4 for SURF-64 or SURF-128. Default: 4 -->
			<value name="iIndexSize" value="4"/>
			<!-- Initial image sampling step. Default: 2 -->
			<value name="iSamplingStep" value="1"/>
			<!-- Number of octaves (orders of magnitude in scale space). Default: 4 -->
			<value name="iOctaves" value="4"/>
			<!-- Lower values result in more generated keypoints. Default:0.2 -->
			<value name="fBlobResponseThreshold" value="0.2"/>
			<!-- Custom lobe size. Default: 3 -->
			<value name="iInitLobeSize" value="3"/>
			<!-- If true, use SURF 128 instead of SURF 64. Default: 0 -->
			<value name="bExtended" value="0"/>
			<!-- number of threads for extraction (0 = use the number of physical cores) -->
			<value name="iNumThreads" value="0"/>

		</Surf>

	</KeyPointExtraction>

</default>


<drinks>

	<ObjectRecognition>
			<!-- Objects to load at startup -->
                        <value name="sLoadObjects" value="bottle"/>
	</ObjectRecognition>

</drinks>



</MergedConfig>